src/main/java/com/microsoft/aad/adal4j/AuthenticationAuthority.java

```diff
@@ -62,12 +63,27 @@
     private final URL authorityUrl;
     private final boolean validateAuthority;
 
+    private final Proxy proxy;
+
     AuthenticationAuthority(final URL authorityUrl,
             final boolean validateAuthority) {
 
         this.authorityUrl = authorityUrl;
         this.authorityType = detectAuthorityType();
         this.validateAuthority = validateAuthority;
+        proxy = Proxy.NO_PROXY;
+        validateAuthorityUrl();
+        setupAuthorityProperties();
+    }
+
+    AuthenticationAuthority(final URL authorityUrl,

```

Comment:
Please expose proxy as a setter property of the instance instead of adding constructor. We want to avoid constructor bloating over a period of time.